cloudinit/apport.py

```diff
@@ -110,30 +116,57 @@ def attach_cloud_info(report, ui=None):
                 report["CloudName"] = "None"
 
 
+def attach_installer_files(report, ui=None):
+    """Attach any subiquity installer logs config.
+
+    To support decoupling apport integration from installer config/logs,
+    we eventually want to either source this function or APPORT_FILES
+    attribute from subiquity  and/or ubuntu-desktop-installer package-hooks
+    python modules.
+    """
+    for apport_file in INSTALLER_APPORT_FILES:
+        attach_file_if_exists(report, apport_file.path, apport_file.label)
+
+
 def attach_user_data(report, ui=None):
     """Optionally provide user-data if desired."""
     if ui:
         user_data_file = _get_user_data_file()
         prompt = (
-            "Your user-data or cloud-config file can optionally be provided"
-            " from {0} and could be useful to developers when addressing this"
-            " bug. Do you wish to attach user-data to this bug?".format(
-                user_data_file
-            )
+            "Your user-data, cloud-config or autoinstall files can optionally "
+            " be provided from {0} and could be useful to developers when"
+            " addressing this bug. Do you wish to attach user-data to this"
+            " bug?".format(user_data_file)
         )
         response = ui.yesno(prompt)
         if response is None:
             raise StopIteration  # User cancelled
         if response:
             attach_file(report, user_data_file, "user_data.txt")
+            for apport_file in INSTALLER_APPORT_SENSITIVE_FILES:
+                attach_file_if_exists(
+                    report, apport_file.path, apport_file.label
+                )
 
 
 def add_bug_tags(report):
     """Add any appropriate tags to the bug."""
+    new_tags = []
+    if report.get("CurtinError"):
+        new_tags.append("curtin")
```

Comment:
Should I be expecting to see these additional tags in LP: #1993011 ?

Reply:
Added context too on the test bug filed.

As this bug was filed from a "live server" install, and there were no curtin-specific errors, we only expect to see the `subiquity` tag.

`curtin` tag would be added if there were a top-level `CurtinError` report key due to curtin specific errors and the presence of `/var/log/installer/curtin-error-logs.tar`).

The `ubuntu-desktop-installer` tag would be added only if ubuntu-bug cloud-init were called from a live desktop installer environment which will add a report key of `UdiLog`. This key is added if the following log exists on the installed system: `/var/log/installer/ubuntu_desktop_installer.log`